database/user.go

```diff
@@ -18,8 +26,147 @@ type (
 	User struct {
 		// ID is a hexadecimal string representation of the MongoDB id assigned
 		// to this user object. It is auto-generated by Mongo on insert.
-		ID   primitive.ObjectID `bson:"_id,omitempty" json:"-"`
-		Sub  string             `bson:"sub" json:"sub"`
-		Tier int                `bson:"tier" json:"tier"`
+		ID          primitive.ObjectID `bson:"_id,omitempty" json:"-"`
+		Sub         string             `bson:"sub" json:"sub"`
+		Tier        int                `bson:"tier" json:"tier"`
+		StorageUsed int                `bson:"storage_used" json:"storageUsed"`
 	}
 )
+
+// UserBySub returns the user with the given sub or nil. The sub is the Kratos
+// id of that user.
+func (db *DB) UserBySub(ctx context.Context, sub string) (*User, error) {
+	users, err := db.managedUsersByField(ctx, "sub", sub)
+	if err != nil {
+		return nil, err
+	}
+	// Subs must be unique. If we hit this then we have a serious
+	// programmer error which endangers customer's data and finances.
+	// We should error out in order to prevent exposing the wrong user data.
+	if len(users) > 1 {
+		build.Critical(fmt.Sprintf("More than one user found for sub '%s'!", sub))
```

Comment:
we should be enforcing this on database level using a unique index on the `sub` field

Reply:
You're right - we should. And we are. I'll remove this check.